brave_domains/service_domains.cc

```diff
@@ -59,42 +60,63 @@ void MaybeWarnSwitchValue(std::string key, std::string value) {
   }
 }
 
+std::string ConvertEnvironmentToString(brave_domains::ServicesEnvironment env) {
+  static const base::flat_map<ServicesEnvironment, std::string> envMap = {
+      {brave_domains::ServicesEnvironment::DEV, kBraveServicesSwitchValueDev},
+      {brave_domains::ServicesEnvironment::STAGING,
+       kBraveServicesSwitchValueStaging},
+      {brave_domains::ServicesEnvironment::PROD,
+       kBraveServicesSwitchValueProduction}};
+
+  auto it = envMap.find(env);
+  if (it != envMap.end()) {
+    return it->second;
+  }
+
+  return kBraveServicesSwitchValueProduction;
```

Comment:
nit: DCHECK or NOTREACHED here, because we shouldn't be in this point as we have every value of the enum covered